test/parachain/src/service.rs

```diff
@@ -76,40 +82,87 @@ macro_rules! new_full_start {
 		}};
 }
 
-/// Run the collator with the given `config`.
+/// Run a collator node with the given parachain `Configuration` and relaychain `Configuration`
+///
+/// This function blocks until done.
 pub fn run_collator<E: sc_service::ChainSpecExtension>(
-	config: Configuration<GenesisConfig, E>,
-	key: Arc<polkadot_primitives::parachain::CollatorPair>,
+	mut parachain_config: Configuration<GenesisConfig, E>,
+	key: Arc<CollatorPair>,
 	polkadot_config: polkadot_collator::Configuration,
-) -> Result<(), sc_cli::error::Error> {
-	let (builder, inherent_data_providers) = new_full_start!(config);
-	inherent_data_providers
-		.register_provider(sp_timestamp::InherentDataProvider)
-		.unwrap();
-
-	let service = builder
-		.with_network_protocol(|_| Ok(NodeProtocol::new()))?
-		.build()?;
-	let proposer_factory = sc_basic_authorship::ProposerFactory {
-		client: service.client(),
-		transaction_pool: service.transaction_pool(),
-	};
-
-	let block_import = service.client();
-
-	let setup_parachain = SetupParachain {
-		service,
-		inherent_data_providers,
-		proposer_factory,
-		block_import,
-	};
-
-	cumulus_collator::run_collator(
-		setup_parachain,
-		crate::PARA_ID,
-		key,
-		polkadot_config,
-	)
+) -> sc_cli::error::Result<()> {
+	let para_id = crate::PARA_ID;
+
+	if polkadot_config.expect_chain_spec().is_kusama() {
```

Comment:
No, this check should not be here. This needs to be done in Polkadot!

Reply:
I'm working on this but it is complicated because the services can be full or light and the signature is different, etc...

MEANWHILE: this was intended :grin: I know you don't like it but the code is also on polkadot and doesn't change. The only reason I duplicate it here is to be able to clone the `task_executor` upon launch.

Reply:
I know that we have the same code in Polkadot, but that is already some generic nightmare there. Downstream should not be required to do the same :)

Reply:
I made the issue #56 to not lose track of this